tests/PullRequest2816.py

```diff
@@ -0,0 +1,53 @@
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2019 Olof-Joachim Frahm <olof@macrolet.net>                        #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+import pytest
+
+from github import GithubException
+
+from . import Framework
+
+
+class PullRequest2816(Framework.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.repo = self.g.get_repo("PyGithub/PyGithub")
+        self.pull = self.repo.get_pull(31)
+
+    def testMergeWithAutomerge(self):
+        self.pull.enable_automerge()
+
+    def testMergeWithAutomergeError(self):
+        with pytest.raises(GithubException) as error:
+            self.pull.enable_automerge()
```

Comment:
Please note down how you recorded the replay data for these two tests given they both call into `self.pull.enable_automerge()`. I presume, your recorded them individually, having auto-merge allowed while recording the first test and disallowed while recording this test. Such a note helps to reproduce the replay data if needed one day.

Reply:
You presumed correctly.
I write some notes on how I generate the data for those tests, I hope that's good enough.